hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -337,6 +338,26 @@ public Partitioner getPartitioner(DataSet<LocatedRow> dataSet, ExecRow template,
     }
 
     @Override
+    public StructType getExternalFileSchema(String storedAs, String location){
+        StructType schema = null;
+        if (storedAs!=null) {
+            if (storedAs.toLowerCase().equals("p")) {
+                schema =  SpliceSpark.getSession().read().parquet(location).schema();
+
+            }
+            if (storedAs.toLowerCase().equals("o")) {
+                schema =  SpliceSpark.getSession().read().orc(location).schema();
+
+            }
+            if (storedAs.toLowerCase().equals("t")) {
+                schema =  SpliceSpark.getSession().read().csv(location).schema();
+            }
+        }
+
+        return schema;
+    }
+
+    @Override
```

Comment:
I know I created this case statement all over the code where we check "pot".  OK, no laughing.  Would you mind encapsulating this a bit so we can lose the checks all over the code?  Maybe an external file interface which implements getExternalFileSchema for example.  I think I have these in a couple of places in the code.